packages/build-utils/src/types.ts

```diff
@@ -320,17 +320,29 @@ export interface BuilderFunctions {
 }
 
 export interface ProjectSettings {
-  framework?: string | null;
+  autoExposeSystemEnvs?: boolean;
+  buildCommand?: string | null;
+  commandForIgnoringBuildStep?: string | null;
+  createdAt?: number;
   devCommand?: string | null;
+  directoryListing?: boolean;
+  framework?: string | null;
+  gitForkProtection?: boolean;
   installCommand?: string | null;
-  buildCommand?: string | null;
+  name?: string;
+  nodeVersion?: '10.x' | '12.x' | '14.x';
   outputDirectory?: string | null;
+  passwordProtection?: {
+    deploymentType: 'all' | 'preview';
+    password?: string | null;
+  };
+  publicSource?: boolean | null;
   rootDirectory?: string | null;
-  createdAt?: number;
-  autoExposeSystemEnvs?: boolean;
+  serverlessFunctionRegion?: string | null;
   sourceFilesOutsideRootDirectory?: boolean;
-  directoryListing?: boolean;
-  gitForkProtection?: boolean;
+  ssoProtection?: {
+    deploymentType: 'all' | 'preview';
+  };
```

Comment:
Do we need two types now? One to represent overridable project settings and one to represent all setting on the project?